File: ЛР1.cs
using System;

enum Zhanr
{
    Adventure,
    Sports,
    Simulator,
    Action,
    Strategy,
}

struct Date
{
    public int Day;
    public int Month;
    public int Year;

    public Date(int day, int month, int year)
    {
        Day = day;
        Month = month;
        Year = year;
    }

    public override string ToString()
    {
        return $"{Day:D2}.{Month:D2}.{Year}";
    }
}

struct Game
{
    public string Title;
    public Zhanr GameGenre;
    public string Publisher;
    public Date PurchaseDate;
    public bool IsCompleted;

    public Game(string title, Zhanr genre, string publisher, Date date, bool completed)
    {
        Title = title;
        GameGenre = genre;
        Publisher = publisher;
        PurchaseDate = date;
        IsCompleted = completed;
    }

    public void PrintInformation()
    {
        Console.WriteLine($"  Тайтл: {Title}");
        Console.WriteLine($"  Разработчик: {Publisher}");
        Console.WriteLine($"  Жанр: {GameGenre}");
        Console.WriteLine($"  Пройдена: {(IsCompleted ? "ДА" : "НЕТ")}");
        Console.WriteLine($"  Дата покупки: {PurchaseDate}");
        Console.WriteLine("+-----------------------------------------+");
    }
}

class Program
{
    static void Main()
    {
        Game[] games = new Game[]
        {

            new Game("Cybeprunk 2077", Zhanr.Action, "CD Projekt Red", new Date(15, 6, 2020), true),

            new Game("FOOTBALL", Zhanr.Sports, "EA Sports", new Date(1, 10, 2021), false),

            new Game("BIOSHOK", Zhanr.Action, "2K Games", new Date(20, 5, 2019), false),

            new Game("GTA VI", Zhanr.Action, "RockstAr", new Date(10, 3, 2018), true),

            new Game("FORZA HORIZON 5", Zhanr.Simulator, "Microsoft", new Date(18, 8, 2020), false)

        };
        Console.WriteLine("  * Игры, которые предстоит пройти:\n");
        foreach (var game in games)
        {
            if (!game.IsCompleted)
            {
                game.PrintInformation();
            }
        }
    }
}

File: ЛР1.cpp
#include <iostream>
#include <string>


#include <locale>

using namespace std;


enum Zhanr
{
    Adventure,
    Sports,
    Simulator,
    Action,
    Strategy
};

struct Date
{
    int Day;
    int Month;
    int Year;

    Date(int day = 1, int month = 1, int year = 2000)
    {
        Day = day;
        Month = month;
        Year = year;
    }

    string ToString() const
    {
        char buffer[20];
        sprintf_s(buffer, "%02d.%02d.%d", Day, Month, Year);
        return string(buffer);
    }
};

struct Game
{
    string Title;
    Zhanr GameGenre;
    string Publisher;
    Date PurchaseDate;
    bool IsCompleted;

    Game(string title, Zhanr genre, string publisher, Date date, bool completed)
    {
        Title = title;
        GameGenre = genre;
        Publisher = publisher;
        PurchaseDate = date;
        IsCompleted = completed;
    }

    void PrintInformation() const
    {
        cout << "  Тайтл: " << Title << endl;
        cout << "  Разработчик: " << Publisher << endl;
        cout << "  Жанр: " << GameGenreToString(GameGenre) << endl;
        cout << "  Пройдена: " << (IsCompleted ? "ДА" : "НЕТ") << endl;
        cout << "  Дата покупки: " << PurchaseDate.ToString() << endl;
        cout << "+-----------------------------------------+" << endl;
    }

    static string GameGenreToString(Zhanr genre)
    {
        switch (genre)
        {
        case Adventure: return "Adventure";
        case Sports: return "Sports";
        case Simulator: return "Simulator";
        case Action: return "Action";
        case Strategy: return "Strategy";
        default: return "Unknown";
        }
    }
};

int main()
{
    setlocale(LC_ALL, "Russian");

    Game games[] = {
        Game("Cybeprunk 2077", Action, "CD Projekt Red", Date(15, 6, 2020), true),
        Game("FOOTBALL", Sports, "EA Sports", Date(1, 10, 2021), false),
        Game("BIOSHOK", Action, "2K Games", Date(20, 5, 2019), false),
        Game("GTA VI", Action, "RockstAr", Date(10, 3, 2018), true),
        Game("FORZA HORIZON 5", Simulator, "Microsoft", Date(18, 8, 2020), false)
    };

    cout << "  * Игры, которые предстоит пройти:\n" << endl;

    for (const auto& game : games)
    {
        if (!game.IsCompleted)
        {
            game.PrintInformation();
        }
    }

    return 0;
}